### PR TITLE
Upgrade to webpack@5

### PR DESCRIPTION
**Pull request by ixrock**
- use `webpack@5` and all related loaders with latest versions
- use external `webpack-dev-server` proxying to `lens-proxy`
- use `@pmmmwh/react-refresh-webpack-plugin` with `react-refresh-typescript` to support HMR

revived from #4725 + #4867

### CODE REVIEW

**Comment by ixrock** [comment]
@lensapp/lens-maintainers PTAL, ready for review again. 
@lensapp/lens-ide build seems to work fine (tested with `LENS_BRANCH=wepback@5-upgrade make -f Makefile.ide build-ide`)

**Review by aleksfront — APPROVED** [review]
LGTM + font issues is fixed!

File: webpack.main.ts [hunk 1]
```diff
@@ -6,23 +6,23 @@
...
-import ProgressBarPlugin from "progress-bar-webpack-plugin";
 import * as vars from "./src/common/vars";
 import getTSLoader from "./src/common/getTSLoader";
 import CircularDependencyPlugin from "circular-dependency-plugin";
+import { iconsAndImagesWebpackRules } from "./webpack.renderer";
 
 const configs: { (): webpack.Configuration }[] = [];
 
 configs.push((): webpack.Configuration => {
   console.info("WEBPACK:main", vars);
```
Comment (Nokel81) [inline webpack.main.ts:18]: Can you please fix this and the `"WEBPACK:renderer"` one too, just change it to be `{ ...vars }`

**Review by Nokel81 — APPROVED** [review]

**Comment by Nokel81** [comment]
I'll make a follow up PR to fix that minor issue.

**Pull request merged**